zookeeper-server/src/main/java/org/apache/zookeeper/server/Request.java

```diff
@@ -78,7 +79,29 @@ public Request(long sessionId, int xid, int type, TxnHeader hdr, Record txn, lon
 
     public final int type;
 
-    public final ByteBuffer request;
+    private final ByteBuffer request;
```

Comment:
This field will later be encapsulated in a `RequestSupplier` similar to @Randgalt done for `RequestPacker` here https://github.com/Randgalt/zookeeper/compare/wip-jute-rpc-generate...wip-jute-rpc-encapsulate.